From 16f5e74cea484db82d983618953e0b3dc8f4cbe3 Mon Sep 17 00:00:00 2001
From: Chad Dupuis <cdupuis@redhat.com>
Date: Thu, 21 Aug 2014 13:05:54 -0400
Subject: [ethernet] qlcnic: Enable IPv6 LRO even if IP address is not programmed

Message-id: <1408626422-3941-71-git-send-email-cdupuis@redhat.com>
Patchwork-id: 88592
O-Subject: [PATCH RHEL 7.1 070/138] qlcnic: Enable IPv6 LRO even if IP address is not programmed
Bugzilla: 1089352
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Shahed Shaikh <shahed.shaikh@qlogic.com>

Bug 1089352 (https://bugzilla.redhat.com/show_bug.cgi?id=1089352)

net commit id e4f8bf830a55c3b5f11cdb836afb32a9a541a116

o Enabling BIT_9 while configuring hardware LRO allows adapter to
  perform LRO even if destination IP address is not programmed in adapter.

Signed-off-by: Shahed Shaikh <shahed.shaikh@qlogic.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/qlogic/qlcnic/qlcnic_hw.c b/drivers/net/ethernet/qlogic/qlcnic/qlcnic_hw.c
index 8a63a3b..3dd161f 100644
--- a/drivers/net/ethernet/qlogic/qlcnic/qlcnic_hw.c
+++ b/drivers/net/ethernet/qlogic/qlcnic/qlcnic_hw.c
@@ -782,8 +782,8 @@ void qlcnic_82xx_config_intr_coalesce(struct qlcnic_adapter *adapter)
    "Could not send interrupt coalescing parameters\n");
 }
 
-#define QLCNIC_ENABLE_IPV4_LRO  1
-#define QLCNIC_ENABLE_IPV6_LRO  2
+#define QLCNIC_ENABLE_IPV4_LRO  BIT_0
+#define QLCNIC_ENABLE_IPV6_LRO  (BIT_1 | BIT_9)
 
 int qlcnic_82xx_config_hw_lro(struct qlcnic_adapter *adapter, int enable)
 {
-- 
1.7.1